### DISCUSSION TITLE
Best way to learning how to use OpenFast?

### DISCUSSION THREAD

**Discussion by unistudent2018** [OpenFAST/openfast#1339] [General]
Hello, after a long time doing the wrong thing trying to run openfast I think I now have a very small clue of how to get it to work. but are there any resources or ways to learn how to use OpenFast? Or any resources?

**Comment by unistudent2018** [comment]
I thought it better to ask for help here, rather than start a new thread.

So I can get OpenFast to run the 5MW_OC4Semi_linear.
![image](https://user-images.githubusercontent.com/117730294/203150117-b3a438cd-6141-46d0-b4c6-cf2931d70ab5.png)
So that seems to be working as it should, although I did notice that it says:
"Running ElastoDyn.
Nodal outputs section of ElastoDyn input file not found or improperly formatted."

Now when I try to run the IEA-15-240-RWT-UMaineSemi I get this:
![image](https://user-images.githubusercontent.com/117730294/203150632-9e63f14c-2e28-4c56-b710-dcd92554f4b1.png)
![image](https://user-images.githubusercontent.com/117730294/203150677-86e1aa1e-3158-4f4e-bfbb-4a780945e788.png)

What am I doing wrong here? Is it anything obvious?

**Reply by jjonkman** [reply]
Dear @unistudent2018,

Regarding learning how to use OpenFAST, a good way to learn is to review the presentations and sample models/simulations that were presented at our last modeling workshops, which are available here: https://drive.google.com/drive/folders/1BDDfcnIyvmZCwf7eFo0ISI7aF_FMAOvt?usp=share_link. Our most recent workshop from September, 2022 (220919) includes presentations, sample models, and video recordings: https://drive.google.com/drive/folders/1Z6hqCxGP57nZTgyTIMi6AkYd83rJuZmh?usp=share_link.

After that, I suggest that you study the various user’s guides and theory manuals for detailed information. There is good documentation in readthedocs (https://openfast.readthedocs.io/en/master/index.html), including details on installing/compiling, running the test suite, using git, and user and theory documentation.

Regarding the "nodal outputs" warnings, these are just warnings and will not effect the simulation results; you can ignore the warnings. A similar question was asked and answered before--see https://github.com/OpenFAST/openfast/issues/573.

Regarding the invalid "requested output channel" warnings, it looks like your HydroDyn write output names are not compatible with the version of OpenFAST you are running.  The names of these HydroDyn outputs were changed in OpenFAST v2.6 and newer, e.g., `WavesF1xi` is now `B1WvsF1xi` for the first potential-flow body.  See the full list of available output channel names for the newest version of OpenFAST here: https://openfast.readthedocs.io/en/main/_downloads/3f19498a5dc774461e022b671ff01ec6/OutListParameters.xlsx.

Regarding the MoorDyn error you are receiving ("unable to parse Line"), it sounds like your MoorDyn input file is not formatted properly.  The MoorDyn input file was changed with the release of MoorDyn v2 in OpenFAST v3.3.  See the input file change documentation for more information: https://openfast.readthedocs.io/en/main/source/user/api_change.html.

Best regards,
